example_external_provider_plugin/plugin.go

```diff
@@ -0,0 +1,215 @@
+	return &provider_service.GetInstanceRootVolumeResult{
+		Volume: &provider_service.Volume{
+			Id:       "volume1",
+			Location: "region1",
+		},
+	}, nil
+}
+
+func (p *Provider) WaitForInstanceReady(context.Context, *provider_service.WaitForInstanceReadyParams) (*provider_service.WaitForInstanceReadyResult, error) {
+	return &provider_service.WaitForInstanceReadyResult{}, nil
+}
+
+func (p *Provider) DeleteInstance(context.Context, *provider_service.DeleteInstanceParams) (*provider_service.DeleteInstanceResult, error) {
+	return &provider_service.DeleteInstanceResult{}, nil
+}
+
+func (p *Provider) AttachVolumeToInstance(context.Context, *provider_service.AttachVolumeToInstanceParams) (*provider_service.AttachVolumeToInstanceResult, error) {
+	return &provider_service.AttachVolumeToInstanceResult{}, nil
+}
+
+func (p *Provider) CopySnapshot(context.Context, *provider_service.CopySnapshotParams) (*provider_service.CopySnapshotResult, error) {
+	return &provider_service.CopySnapshotResult{
+		Snapshot: &provider_service.Snapshot{
+			Id:       "snapshot1",
+			Location: "region2",
+		},
+	}, nil
+}
+
+func (p *Provider) DeleteSnapshot(context.Context, *provider_service.DeleteSnapshotParams) (*provider_service.DeleteSnapshotResult, error) {
+	return &provider_service.DeleteSnapshotResult{}, nil
+}
+
+func (p *Provider) WaitForSnapshotReady(context.Context, *provider_service.WaitForSnapshotReadyParams) (*provider_service.WaitForSnapshotReadyResult, error) {
+	return &provider_service.WaitForSnapshotReadyResult{}, nil
+}
+
+func (p *Provider) CreateVolumeFromSnapshot(context.Context, *provider_service.CreateVolumeFromSnapshotParams) (*provider_service.CreateVolumeFromSnapshotResult, error) {
+	return &provider_service.CreateVolumeFromSnapshotResult{
+		Volume: &provider_service.Volume{
+			Id:       "volume2",
+			Location: "region2",
+		},
+	}, nil
+}
+
+func (p *Provider) TakeVolumeSnapshot(context.Context, *provider_service.TakeVolumeSnapshotParams) (*provider_service.TakeVolumeSnapshotResult, error) {
+	return &provider_service.TakeVolumeSnapshotResult{
+		Snapshot: &provider_service.Snapshot{
+			Id:       "snapshot11",
+			Location: "region1",
+		},
+	}, nil
+}
+
+func (p *Provider) WaitForVolumeReady(context.Context, *provider_service.WaitForVolumeReadyParams) (*provider_service.WaitForVolumeReadyResult, error) {
+	return &provider_service.WaitForVolumeReadyResult{}, nil
+}
+
+func (p *Provider) WaitForVolumeAttached(context.Context, *provider_service.WaitForVolumeAttachedParams) (*provider_service.WaitForVolumeAttachedResult, error) {
+	return &provider_service.WaitForVolumeAttachedResult{}, nil
+}
+
+func (p *Provider) DeleteVolume(context.Context, *provider_service.DeleteVolumeParams) (*provider_service.DeleteVolumeResult, error) {
+	return &provider_service.DeleteVolumeResult{}, nil
+}
+
+//
+//func main() {
```

Comment:
Please add the main back and add a small readme of how to build and setup the example provider as an External provider to the backend